bin/varnishd/cache/cache_backend.h

```diff
@@ -44,34 +44,50 @@ struct vrt_ctx;
 struct vrt_backend_probe;
 struct conn_pool;
 
+/*--------------------------------------------------------------------*/
+
+enum connwait_e {
+	CW_DO_CONNECT = 1,
+	CW_QUEUED,
+	CW_BE_BUSY,
+};
+
+struct backend_cw {
+	unsigned			magic;
+#define BACKEND_CW_MAGIC		0x75c7a52b
+	enum connwait_e			cw_state;
+	VTAILQ_ENTRY(backend_cw)	cw_list;
+	pthread_cond_t			cw_cond;
+};
```

Comment:
I think you should be able to get away with just an opaque `struct connwait;` here and keep the actual type definitions contained in `cache_backend.c`.

Please note how I'm implicitly suggesting to rename the struct :grin:

Reply:
Moved the struct to `cache_backend.c` and renamed as suggested.